// coininfo/lib/coins/phr.js
/*
  info from:
    https://github.com/phoreproject
*/

var common = {
  name: 'Phore',
  per1: 1e8,
  unit: 'PHR'
}

var main = Object.assign({}, {
  hashGenesisBlock: '0x2b1a0f66712aad59ad283662d5b919415a25921ce89511d73019107e380485bf',
  // nDefaultPort
  port: 11771,
  portRpc: 11772,
  protocol: {
    // pchMessageStart
    magic: 0x91c4fde9 // careful, sent over wire as little endian
  },
  // vSeeds
  seedsDns: [
    "dns0.phore.io",
    "phore.seed.rho.industries"
  ],
  // base58Prefixes
  versions: {
    bip32: {
      private: 0x05358394,
      public: 0x022D2533
    },
    bip44: 444,
    private: 0x80,
    public: 55,
    scripthash: 13
  }
}, common)

var test = Object.assign({}, {
  hashGenesisBlock: '2b1a0f66712aad59ad283662d5b919415a25921ce89511d73019107e380485bf',
  port: 11773,
  portRpc: 11774,
  protocol: {
    magic: 0x477665ba
  },
  seedsDns: [
    "pivx-testnet.seed.fuzzbawls.pw",
    "pivx-testnet.seed2.fuzzbawls.pw",
    "s3v3nh4cks.ddns.net"
  ],
  versions: {
    bip32: {
      private: 0x05358394,
      public: 0x022D2533
    },
    bip44: 444,
    private: 0xef,
    public: 139,
    scripthash: 19
  }
}, common)

var regtest = Object.assign({}, {
  hashGenesisBlock: '2b1a0f66712aad59ad283662d5b919415a25921ce89511d73019107e380485bf',
  port: 11773,
  portRpc: 11774,
  protocol: {
    magic: 0x477665ba
  },
  seedsDns: [],
  versions: {
    bip32: {
      private: 0x05358394,
      public: 0x022D2533
    },
    bip44: 444,
    private: 0xef,
    public: 139,
    scripthash: 19
  }
}, common)

module.exports = {
  main: main,
  test: test,
  regtest: regtest
}
